doc/cmp.txt

```diff
@@ -321,7 +321,42 @@ Highlight                                                        *cmp-highlight*
 *CmpItemMenu*
   The menu field's highlight group.
 
```

Comment:
I think separating `CmpCompletionWindow` and `CmpDocumentationWindow` is good idea 👍🏼 

But I can't understand what the `CmpWindow` means, could you explain me?

Reply:
While `CmpCompletionWindow` or `CmpDocumentationWindow` refer to the completion and documentation windows respectively, `CmpWindow` refers to both.

Some of the highlight groups were too different to link to one `CmpWindow` variant (for example, `CmpCompletionWindow` had to be `Pmenu` and `CmpDocumentationWindow` was `NormalFloat`). Other groups like `CmpCompletionWindow` and `CmpDocumentationBorder` didn't need to be different, so I made a common `CmpWindowBorder` so it would be easier for users to change both at once.

Reply:
I can understand but it's a bit complicated...

In my PR, I tried to implement it with the following idea. What do you think of this idea?

```lua
cmp.setup {
  window = {
    completion = {
      winhighlight = {
        bordered = 'NormalFloat:CmpBorderedWindow,FloatBorder:CmpBoerededWindowBorder,CursorLine:Visual,Search:None',
        defaults = 'NormalFloat:Pmenu,FloatBorder:Pmenu,CursorLine:PmenuSel,Search:None',
      }
    },
    {
winhighlight = {
        bordered = 'NormalFloat:CmpBorderedWindow,FloatBorder:CmpBoerededWindowBorder,CursorLine:Visual,Search:None',
        defaults = 'FloatBorder:NormalFloat,Search:None',
      }
    }
  }
}

```

The default setting is the above. So the highlight will be automatically changed if the user set the `border = 'single'` as a configuration option.

Reply:
I would prefer setting the highlights in my colorscheme with some highlight group, but that is just my personal opinion😅

Reply:
> I would prefer setting the highlights in my colorscheme with some highlight group, but that is just my personal opinion😅

me too, much cleaner

Reply:
I added both approaches in a recent commit. I think there are benefits on the backend to the approach from hrsh7th, and benefits on the front to the `highlight` approach.